src/main/java/com/atlassian/bitbucket/jenkins/internal/deployments/DeploymentPosterImpl.java

```diff
@@ -25,25 +30,63 @@
 @Singleton
 public class DeploymentPosterImpl implements DeploymentPoster {
 
-    protected static final Logger LOGGER = Logger.getLogger(DeploymentPosterImpl.class.getName());
+    private static final Logger LOGGER = Logger.getLogger(DeploymentPosterImpl.class.getName());
 
     private final BitbucketClientFactoryProvider bitbucketClientFactoryProvider;
+    private final JenkinsProvider jenkinsProvider;
     private final JenkinsToBitbucketCredentials jenkinsToBitbucketCredentials;
     private final BitbucketPluginConfiguration pluginConfiguration;
 
     @Inject
     public DeploymentPosterImpl(BitbucketClientFactoryProvider bitbucketClientFactoryProvider,
+                                JenkinsProvider jenkinsProvider,
                                 JenkinsToBitbucketCredentials jenkinsToBitbucketCredentials,
                                 BitbucketPluginConfiguration pluginConfiguration) {
         this.bitbucketClientFactoryProvider = bitbucketClientFactoryProvider;
+        this.jenkinsProvider = jenkinsProvider;
         this.jenkinsToBitbucketCredentials = jenkinsToBitbucketCredentials;
         this.pluginConfiguration = pluginConfiguration;
     }
 
     @Override
-    public void postDeployment(String bbsServerId, String projectKey, String repositorySlug, String revisionSha,
-                               BitbucketDeployment deployment, Run<?, ?> run,
+    public void onCheckout(Run<?, ?> build, TaskListener listener) {
+        try {
+            BitbucketRevisionAction revisionAction = build.getAction(BitbucketRevisionAction.class);
+            if (revisionAction == null) {
+                // Not a Bitbucket checkout
+                return;
+            }
+            if (!(build instanceof FreeStyleBuild)) {
+                // Not a freestyle build so we can't get the publisher off it
+                return;
+            }
```

Comment:
This won't affect our multibranch pipelines, then, and I won't need to search for a way to suppress it.

Reply:
We do  plan to support multibranch pipelines, but handling so far is just for Freestyle.
Can you expand on what you mean by "search for a way to suppress it"?

Reply:
I mean the build status notification is automatic and cannot be suppressed ([JENKINS-61271](https://issues.jenkins.io/browse/JENKINS-61271)), so I was concerned that this might be adding something similar. But I see the deployment notification needs parameters for which there cannot be universally applicable default values, so it can never be made that automatic.